src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1557,21 +1582,18 @@ private async ValueTask NotifyFeaturesAboutValueChangeAsync(long index, string t
             var taken = false;
             while (taken == false)
             {
-                taken = await _clusterLocker.WaitAsync(TimeSpan.FromSeconds(5));
+                taken = await _updateValuesLocker.WaitAsync(TimeSpan.FromSeconds(5), DatabaseShutdown);
 
                 try
                 {
                     if (CanSkipValueChange(index, type))
                         return;
 
-                    if (DatabaseShutdown.IsCancellationRequested)
-                        return;
+                    DatabaseShutdown.ThrowIfCancellationRequested();
```

Comment:
Just to make sure - will the exception be _properly_ handled by the caller?

Reply:
Yes. This is already handled.